--- Pandas Tutorial ---

- import pandas

In [2]:
import pandas as pd

- verefication des functions dans pandas

In [30]:
dir (pd)

['ArrowDtype',
 'BooleanDtype',
 'Categorical',
 'CategoricalDtype',
 'CategoricalIndex',
 'DataFrame',
 'DateOffset',
 'DatetimeIndex',
 'DatetimeTZDtype',
 'ExcelFile',
 'ExcelWriter',
 'Flags',
 'Float32Dtype',
 'Float64Dtype',
 'Grouper',
 'HDFStore',
 'Index',
 'IndexSlice',
 'Int16Dtype',
 'Int32Dtype',
 'Int64Dtype',
 'Int8Dtype',
 'Interval',
 'IntervalDtype',
 'IntervalIndex',
 'MultiIndex',
 'NA',
 'NaT',
 'NamedAgg',
 'Period',
 'PeriodDtype',
 'PeriodIndex',
 'RangeIndex',
 'Series',
 'SparseDtype',
 'StringDtype',
 'Timedelta',
 'TimedeltaIndex',
 'Timestamp',
 'UInt16Dtype',
 'UInt32Dtype',
 'UInt64Dtype',
 'UInt8Dtype',
 '__all__',
 '__builtins__',
 '__cached__',
 '__doc__',
 '__docformat__',
 '__file__',
 '__git_version__',
 '__loader__',
 '__name__',
 '__package__',
 '__path__',
 '__spec__',
 '__version__',
 '_config',
 '_is_numpy_dev',
 '_libs',
 '_testing',
 '_typing',
 '_version',
 'annotations',
 'api',
 'array',
 'arrays',
 'bdate_range',
 'compat',
 'concat',
 'c

- Vérification de la version de Pandas

In [20]:
print(pd.__version__)

2.0.3


- First example

In [6]:
mydataset = {
  'CARS': ["SKODA", "SEAT", "KIA"],
  'QUANTITY': [5, 4, 3]
}

myvar = pd.DataFrame(mydataset)

print(myvar)

    CARS  QUANTITY
0  SKODA         5
1   SEAT         4
2    KIA         3


- Une série Pandas
Qu'est-ce qu'une série ?
Une série Pandas est comme une colonne dans une table.
C'est un tableau unidimensionnel contenant des données de n'importe quel type.

In [13]:
a = [50 , 42 , 53]

myvar = pd.Series(a)

print(myvar)

0    50
1    42
2    53
dtype: int64


- Libellés
Si rien d'autre n'est spécifié, les valeurs sont étiquetées avec leur numéro d'index. La première valeur a un index 0, la deuxième valeur a un index 1, etc.

Cet libellé peut être utilisé pour accéder à une valeur spécifiée.

In [11]:
print(myvar[3])

50


- Créer des étiquettes
Avec l'argument d'index, vous pouvez attribuer vos propres noms aux étiquettes.

In [23]:
myvar = pd.Series(a, index = ["element1", "element2", "element3"])
print(myvar)
print("element 1 : ",myvar[0])
print("element 1 : ",myvar["element1"])

element1    50
element2    42
element3    53
dtype: int64
element 1 :  50
element 1 :  50


- Objets clé/valeur en tant que série
Vous pouvez également utiliser un objet clé/valeur, tel qu'un dictionnaire, lors de la création d'une série.

In [26]:
calories = {"day1": 420, "day2": 380, "day3": 390}

myvar = pd.Series(calories, index = ["day1", "day2"])

print(myvar)

day1    420
day2    380
dtype: int64


- DataFrames

Qu'est-ce qu'un DataFrame ?

Les ensembles de données dans Pandas sont généralement des tableaux multi-dimensionnels appelés DataFrames.

Une série est similaire à une colonne, et un DataFrame représente l'ensemble de la table.

In [35]:
data = {
    "brand": ["nike", "adidas", "zara"],
    "quantity": [50, 40, 45],
    "price": [200, 350, 800],
}

df = pd.DataFrame(data)

print(df)

    brand  quantity  price
0    nike        50    200
1  adidas        40    350
2    zara        45    800


- Localiser une ligne

Comme vous pouvez le voir à partir du résultat ci-dessus, le DataFrame est semblable à une table avec des lignes et des colonnes.

Pandas utilise l'attribut loc pour retourner une ou plusieurs lignes spécifiées.

In [37]:
print(df.loc[1])

brand       adidas
quantity        40
price          350
Name: 1, dtype: object


- Remarque : Cet exemple renvoie une série Pandas.

In [41]:
print(df.loc[[0 , 2]])

  brand  quantity  price
0  nike        50    200
2  zara        45    800


- Index nommés
Avec l'argument d'index, vous pouvez attribuer des noms à vos propres index.

In [42]:
data = {
    "brand": ["nike", "adidas", "zara"],
    "price": [200, 350, 800],
}

df = pd.DataFrame(data, index = ["brand1", "brand2", "brand3"])

print(df) 

         brand  price
brand1    nike    200
brand2  adidas    350
brand3    zara    800


- Localiser des index nommés
Utilisez l'index nommé dans l'attribut loc pour retourner la ou les lignes spécifiées.

Exemple
Retourner "brand2" :

In [43]:
print(df.loc["brand2"])

brand    adidas
price       350
Name: brand2, dtype: object


- Lecture des fichiers CSV
Une manière simple de stocker de grands ensembles de données est d'utiliser des fichiers CSV (comma separated files).

Les fichiers CSV contiennent du texte brut et sont un format bien connu pouvant être lu par tout le monde, y compris par Pandas.

Dans nos exemples, nous utiliserons un fichier CSV appelé 'data.csv'.

In [44]:
df = pd.read_csv('data.csv')

print(df.to_string())

     Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
5          60    102       127     300.0
6          60    110       136     374.0
7          45    104       134     253.3
8          30    109       133     195.1
9          60     98       124     269.0
10         60    103       147     329.3
11         60    100       120     250.7
12         60    106       128     345.3
13         60    104       132     379.3
14         60     98       123     275.0
15         60     98       120     215.2
16         60    100       120     300.0
17         45     90       112       NaN
18         60    103       123     323.0
19         45     97       125     243.0
20         60    108       131     364.2
21         45    100       119     282.0
22         60    130       101     300.0
23         45   

- max_rows

Le nombre de lignes retournées est défini dans les paramètres d'options de Pandas.

Vous pouvez vérifier le nombre maximum de lignes de votre système avec l'instruction pd.options.display.max_rows.

In [45]:
print(pd.options.display.max_rows) 

60


Augmentez le nombre maximum de lignes pour afficher l'intégralité du DataFrame :

In [46]:
pd.options.display.max_rows = 9999

df = pd.read_csv('data.csv')

print(df) 

     Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
5          60    102       127     300.0
6          60    110       136     374.0
7          45    104       134     253.3
8          30    109       133     195.1
9          60     98       124     269.0
10         60    103       147     329.3
11         60    100       120     250.7
12         60    106       128     345.3
13         60    104       132     379.3
14         60     98       123     275.0
15         60     98       120     215.2
16         60    100       120     300.0
17         45     90       112       NaN
18         60    103       123     323.0
19         45     97       125     243.0
20         60    108       131     364.2
21         45    100       119     282.0
22         60    130       101     300.0
23         45   

- Lecture de fichiers JSON
Les ensembles de données volumineux sont souvent stockés ou extraits sous forme de fichiers JSON.

JSON est du texte brut, mais il a le format d'un objet et est bien connu dans le monde de la programmation, y compris par Pandas.

Dans nos exemples, nous utiliserons un fichier JSON appelé 'data.json'.

Astuce : utilisez to_string() pour imprimer l'intégralité du DataFrame.

In [49]:
import pandas as pd

df = pd.read_json('data.json')

print(df.to_string()) 

     Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
5          60    102       127     300.5
6          60    110       136     374.0
7          45    104       134     253.3
8          30    109       133     195.1
9          60     98       124     269.0
10         60    103       147     329.3
11         60    100       120     250.7
12         60    106       128     345.3
13         60    104       132     379.3
14         60     98       123     275.0
15         60     98       120     215.2
16         60    100       120     300.0
17         45     90       112       NaN
18         60    103       123     323.0
19         45     97       125     243.0
20         60    108       131     364.2
21         45    100       119     282.0
22         60    130       101     300.0
23         45   

Dictionary comme JSON

JSON = Dictionnaire Python

Les objets JSON ont le même format que les dictionnaires Python.

Si votre code JSON n'est pas dans un fichier, mais dans un dictionnaire Python, vous pouvez le charger directement dans un DataFrame :

In [53]:
data = {
  "Duration":{
    "0":60,
    "1":60,
    "2":60,
  },
  "Pulse":{
    "0":110,
    "1":117,
    "2":103,
  },
  "Maxpulse":{
    "0":130,
    "1":145,
    "2":135,
  },
  "Calories":{
    "0":409,
    "1":479,
    "2":340,
  }
}

df = pd.DataFrame(data)

print(df)

   Duration  Pulse  Maxpulse  Calories
0        60    110       130       409
1        60    117       145       479
2        60    103       135       340


- Pandas - Analyse des DataFrames

Visualisation des données

L'une des méthodes les plus utilisées pour obtenir un aperçu rapide du DataFrame est la méthode head().

La méthode head() renvoie les en-têtes et un nombre spécifié de lignes, en commençant par le haut.

exemple : Obtenez un aperçu rapide en imprimant les dix premières lignes du DataFrame :

!!! Note : si le nombre de lignes n'est pas spécifié, la méthode head() renverra les 5 premières lignes par défaut.

In [54]:
import pandas as pd

df = pd.read_csv('data.csv')

print(df.head(10))

   Duration  Pulse  Maxpulse  Calories
0        60    110       130     409.1
1        60    117       145     479.0
2        60    103       135     340.0
3        45    109       175     282.4
4        45    117       148     406.0
5        60    102       127     300.0
6        60    110       136     374.0
7        45    104       134     253.3
8        30    109       133     195.1
9        60     98       124     269.0


Il existe également une méthode tail() pour visualiser les dernières lignes du DataFrame.

La méthode tail() renvoie les en-têtes et un nombre spécifié de lignes, en commençant par le bas.

In [55]:
print(df.tail()) 

     Duration  Pulse  Maxpulse  Calories
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4


- Informations sur les données
L'objet DataFrames dispose d'une méthode appelée info() qui vous fournit davantage d'informations sur l'ensemble de données.

In [56]:
print(df.info()) 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 169 entries, 0 to 168
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Duration  169 non-null    int64  
 1   Pulse     169 non-null    int64  
 2   Maxpulse  169 non-null    int64  
 3   Calories  164 non-null    float64
dtypes: float64(1), int64(3)
memory usage: 5.4 KB
None


- Valeurs nulles
La méthode info() nous indique également combien de valeurs non nulles sont présentes dans chaque colonne, et dans notre ensemble de données, il semble y avoir 164 valeurs non nulles sur 169 dans la colonne "Calories".

Cela signifie qu'il y a 5 lignes sans aucune valeur dans la colonne "Calories", pour une raison quelconque.

Les valeurs vides, ou les valeurs nulles, peuvent poser problème lors de l'analyse des données, et vous devriez envisager de supprimer les lignes avec des valeurs vides. C'est une étape vers ce que l'on appelle le nettoyage des données, et vous en apprendrez davantage à ce sujet dans les chapitres suivants.